pkg/relay/ethereum/tx.go

```diff
@@ -124,14 +122,13 @@ func (c *Chain) SendMsgs(msgs []sdk.Msg) ([]core.MsgID, error) {
 	return msgIDs, nil
 }
 
-func (c *Chain) GetMsgResult(id core.MsgID) (core.MsgResult, error) {
+func (c *Chain) GetMsgResult(ctx context.Context, id core.MsgID) (core.MsgResult, error) {
 	logger := c.GetChainLogger()
 
 	msgID, ok := id.(*MsgID)
 	if !ok {
 		return nil, fmt.Errorf("unexpected message id type: %T", id)
 	}
-	ctx := context.TODO()
 	txHash := msgID.TxHash()
 	receipt, err := c.client.WaitForReceiptAndGet(ctx, txHash)
```

Comment:
@abicky If you want to be ultimately nervous, you should replace this `ctx` with `context.TODO()`, right?